pandas/tests/indexing/test_loc.py

```diff
@@ -2971,6 +2971,28 @@ def test_loc_periodindex_3_levels():
     assert mi_series.loc[(p_index[0], "A", "B")] == 1.0
 
 
+@pytest.mark.parametrize("utc", [False, True])
+def test_loc_datetime_assignment_dtype_does_not_change(utc):
+    # GH#49837
+    df = DataFrame(
+        {
+            # Checking with and without UTC as the original bug didn't occur
+            # for utc=True
+            "date": to_datetime(
+                [datetime(2022, 1, 20), datetime(2022, 1, 22)], utc=utc
+            ),
+            "update": [True, False],
+        }
+    )
+    # We don't expect the dataframe to be changes at all
+    expected = df.copy(deep=True)
+
+    update_df = df[df["update"]]
+    df.loc[df["update"], ["date"]] = update_df["date"]
```

Comment:
Can you test for ``df.loc[df["update"], ["date"]]`` and ``df.loc[df["update"], "date"]``? This takes significantly different code paths.

Reply:
yup, actually the bug didn't occur in the second one.
will add.

BTW - is this the right place for the test?

Reply:
good point, better place is probably frame/indexing/test_indexing